### PR TITLE
first pass at tinyble variant

### PR DESCRIPTION
**Pull request by jacobrosenthal**
Working PR, no rush to merge, comments accepted.

Theres only 4 pins available on the tinyble so Im duping them across all digital, analog, spi, and i2c. Also theres an accelerometer on board but Im not defining that in the variant .. not sure if I should? Have to take a look at how whatever MPU6050 library works

### CODE REVIEW

File: variants/TinyBLE/variant.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+#include "WVariant.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif // __cplusplus
+
+// Number of pins defined in PinDescription array
+#define PINS_COUNT           (32u)
```
Comment (sandeepmistry) [inline variants/TinyBLE/variant.h]: Any reason to expose all the pins even though they aren't broken out?

**Comment by jacobrosenthal** [comment]
Thanks for the code review!

I dont know.. didnt know it mattered I guess?

So it should be all externally interface-able pins. Ive noticed Serial is defined on say, nrf dongle, which only goes through the segger and thus isnt actually broken out and probably shouldnt be incldued right? Nor SPI and I2C?

**Comment by jacobrosenthal** [comment]
Also whenever I get specific the blink doesnt blink.. Is there an ordering here?

**Comment by sandeepmistry** [comment]
Sorry, not PIN_COUNT, but `NUM_DIGITAL_PINS`

I think you should be able to loop through 0 - NUM_DIGITAL_PINS and send HIGH or LOW on some external pins. So those exclude the non-broken out SPI or I2C pins, but could include them if they were broken out to headers.

**Comment by jacobrosenthal** [comment]
Based on my last discussion with sandeep I set NUM_DIGITAL_PINS to 8 (3 LEDS, 1 Button, 4 exposed pins)

File: boards.txt [hunk 2]
```diff
@@ -439,6 +439,48 @@ nRF51Dongle.menu.softdevice.s130.upload.maximum_size=151552
 nRF51Dongle.menu.softdevice.s130.build.extra_flags=-DNRF51 -DS130 -DNRF51_S130
 nRF51Dongle.menu.softdevice.s130.build.ldscript=armgcc_s130_nrf51822_{build.chip}.ld
 
+
+TinyBLE.name= TinyBLE
+
+TinyBLE.upload.tool=sandeepmistry:openocd
+TinyBLE.upload.protocol=cmsis-dap
```
Comment (sandeepmistry) [inline boards.txt]: This depends on #114 right?

File: variants/TinyBLE/variant.h [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+//3 LEDS, 1 Button, 4 exposed pins
+#define NUM_DIGITAL_PINS     (8u)
```
Comment (sandeepmistry) [inline variants/TinyBLE/variant.h]: I think this is good now, to confirm the digital pins are:

21, 22, 23, 17, 3, 4, 5, 6?

Comment (jacobrosenthal) [inline variants/TinyBLE/variant.h]: Yep

File: boards.txt (hunk 2 above)
Comment (jacobrosenthal) [inline boards.txt]: I would argue our default selection has never worked, everyone has manually selected up until now, so it doesnt depend on it, but rather will be useful when 114 lands.

**Comment by sandeepmistry** [comment]
I'll merge this once #114 is merged.

**Pull request merged**